doc/graphene-specification-v2.2.mediawiki

```diff
@@ -0,0 +1,391 @@
+
+How much space is required to relay blocks using Bloom filters? For example, we could set the FPR of the Bloom Filter to ''f=1/m''. In that case, when the receiver checks each of the ''m'' transaction IDs in her mempoool, we can expect that the Bloom filter will wrongly state that ''f*m=(1/m)*m=1'' transaction is in the block on average. To make matters worse, we won't know which transaction is the wrong one; as a result of the extra transaction, the Merkle root won't validate. We can try to fix this problem by lowering the FPR of the filter. For example, if we set the FPR to ''f=1/(144m)'', then we can expect that the filter will permit a wrong answer only about once every 144 blocks relayed (i.e., only about once a day). But keep in mind, this accuracy will cost us in bytes. The size of a Bloom filter with ''n'' items inserted and a false positive rate of ''f=1/144m'' [https://en.wikipedia.org/wiki/Bloom_filter#Optimal_number_of_hash_functions is well known to be] 
+''-n*ln(f)/ln<sup>2</sup>(2) = n*ln(1/(144m))/(8ln<sup>2</sup>(2))'' bytes. For example, for a block with ''n=2000'' transactions and a mempool of ''m=6000'' transactions total, the Bloom filter will be about 7,113 bytes. That's an improvement over our first and second options, but we can do better. 
+
+A fourth option is presented by Invertible Bloom Lookup Tables (IBLTs) [[graphene-specification-v2.2.mediawiki#ref2|<nowiki>[2]</nowiki>]], another useful probabilistic data structure. They are designed to allow us to discover the [https://en.wikipedia.org/wiki/Symmetric_difference symmetric difference] between two sets of items. For example, we can create an IBLT of all transactions IDs that are in the sender's block, and then create another IBLT of the transactions in the receiver's mempool. A ''subtraction'' [[graphene-specification-v2.2.mediawiki#ref3|<nowiki>[3]</nowiki>]] of the first IBLT from the second will tell us exactly which transactions in the mempool are not in the block. Given this functionality, one can use IBLTs alone [[graphene-specification-v2.2.mediawiki#ref7|<nowiki>[7]</nowiki>]] to relay the block from sender to receiver, but unfortunately it is not an efficient approach. The size in bytes of an IBLT increases linearly with the size of the symmetric difference recovered from it. An IBLT uses about 17 bytes per transaction ID that is part of the symmetric difference, and the overhead of an IBLT (in bytes) is about 140% (this value can vary significantly for small symmetric differences, and can be set optimally using techniques described in the Recovery section below). So if the mempool is 2000 transactions larger than the block (i.e., the symmetric difference is 2000), then the sender's IBLT will be about ''(1.4*2000)*17=47,600'' bytes. Not our best option so far.
```

Comment:
"of all transactions IDs that" -> "of all transaction IDs that"

Reply:
Done.